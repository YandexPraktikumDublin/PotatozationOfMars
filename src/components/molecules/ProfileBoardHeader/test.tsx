import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { ProfileBoardHeader } from '.'

describe('<ProfileBoardHeader />', () => {
  it('should renders correct <ProfileBoardHeader />', () => {
    const wrapper = shallow(<ProfileBoardHeader />)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should has className from className prop', () => {
    const className = 'test-class-name'

    const wrapper = shallow(<ProfileBoardHeader className={className} />)

    expect(wrapper.prop('className')).toMatch(className)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
