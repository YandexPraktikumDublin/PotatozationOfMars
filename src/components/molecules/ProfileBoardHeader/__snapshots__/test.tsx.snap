// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileBoardHeader /> should has className from className prop 1`] = `
<header
  className="relative text-center test-class-name"
>
  <ProfileBoardHeaderEditButton
    onClick={[Function]}
  />
  <button
    className="group relative rounded-full border border-white outline-none overflow-hidden mx-auto mb-6"
  >
    <img
      alt=""
      height="80"
      src={[Function]}
      width="80"
    />
    <div
      className="hidden group-hover:flex justify-center items-center absolute inset-0 bg-primary bg-opacity-50 text-white"
      onClick={[Function]}
    >
      change
    </div>
  </button>
  <h1
    className="font-medium"
  >
    Ivan Ivanov
  </h1>
</header>
`;

exports[`<ProfileBoardHeader /> should renders correct <ProfileBoardHeader /> 1`] = `
<header
  className="relative text-center"
>
  <ProfileBoardHeaderEditButton
    onClick={[Function]}
  />
  <button
    className="group relative rounded-full border border-white outline-none overflow-hidden mx-auto mb-6"
  >
    <img
      alt=""
      height="80"
      src={[Function]}
      width="80"
    />
    <div
      className="hidden group-hover:flex justify-center items-center absolute inset-0 bg-primary bg-opacity-50 text-white"
      onClick={[Function]}
    >
      change
    </div>
  </button>
  <h1
    className="font-medium"
  >
    Ivan Ivanov
  </h1>
</header>
`;
